pkg/crc/pullsecret/pullsecret.go

```diff
@@ -0,0 +1,176 @@
+	if err != nil {
+		return err
+	}
+	logging.DebugF("Output is : %s", output)
+	return nil
+}
+
+func addpullSecretToIntanceDisk(driver drivers.Driver, pullSec string) error {
+	output, err := drivers.RunSSHCommandFromDriver(driver, fmt.Sprintf("cat <<EOF | sudo tee /var/lib/kubelet/config.json\n%s\nEOF", pullSec))
+	if err != nil {
+		return err
+	}
+	logging.DebugF("Output is : %s", output)
+	return nil
+}
+
+func addKubeconfigFileToInstance(driver drivers.Driver, kubeconfigFilePath string) error {
+	kubeconfig, err := ioutil.ReadFile(kubeconfigFilePath)
+	if err != nil {
+		return err
+	}
+	_, err = drivers.RunSSHCommandFromDriver(driver, fmt.Sprintf("cat <<EOF | tee /tmp/kubeconfig\n%s\nEOF", string(kubeconfig)))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func replaceUserPullSecret(driver drivers.Driver) error {
+	output, err := drivers.RunSSHCommandFromDriver(driver, replacePullSecretCmd)
+	if err != nil {
+		return err
+	}
+	logging.DebugF("Output of %s: %s", replacePullSecretCmd, output)
+	return nil
+}
+
+func addClusterID(driver drivers.Driver) error {
+	clusterID := uuid.New()
+	updateClusterIdCmd := fmt.Sprintf(updateClusterIdCmd, clusterID)
+	output, err := drivers.RunSSHCommandFromDriver(driver, updateClusterIdCmd)
+	if err != nil {
+		return err
+	}
+	logging.DebugF("Output of %s: %s", updateClusterIdCmd, output)
+	return nil
+}
+
+func setPullSecretAndClusterID(driver drivers.Driver) (rerr error) {
+	m := errors.MultiError{}
+	sd := systemd.NewInstanceSystemdCommander(driver)
+	defer func() {
+		// Stop the kubelet service.
+		if _, err := sd.Stop("kubelet"); err != nil {
+			m.Collect(err)
+		}
+		output, err := drivers.RunSSHCommandFromDriver(driver, stopAndRemovePodsCmd)
+		logging.DebugF("Output of %s: %s", stopAndRemovePodsCmd, output)
+		if err != nil {
+			m.Collect(err)
+		}
+		rerr = m.ToError()
+	}()
+
+	// Start the kubelet service
+	if _, err := sd.Start("kubelet"); err != nil {
+		m.Collect(err)
+	}
+
+	// Replace existing pull secret with the user pull secret
+	if err := replaceUserPullSecret(driver); err != nil {
+		m.Collect(err)
+	}
+
+	// add random cluster id
+	if err := addClusterID(driver); err != nil {
+		m.Collect(err)
+	}
+
+	return nil
+}
+
+func StorePullSecretToFile(pullsecret string, destDir string, fileName string, mode os.FileMode) error {
+	err := os.MkdirAll(destDir, mode)
+	if err != nil && !os.IsExist(err) {
+		return fmt.Errorf("Cannot create directory %s: %v", destDir, err)
+	}
+	pullSecretFilePath := filepath.Join(destDir, fileName)
+	// Use 0600 mode to create this so it is readable only by current user.
+	if err := ioutil.WriteFile(pullSecretFilePath, []byte(pullsecret), 0600); err != nil {
+		return fmt.Errorf("Cannot write to %s: %v", pullSecretFilePath, err)
+	}
+	return nil
+}
+
+func GetPullSecretFromFile(pullSecretFilePath string) (string, bool, error) {
```

Comment:
`GetPullSecretFromFile` and `StorePullSecretToFile` are not consistent, one takes dir + filename, the other takes full path.

Reply:
Maybe https://github.com/cfergeau/crc/commit/b5fd6743e20e71b6766adb7c91835bf11e7c8f41 ?

Reply:
@cfergeau that just matter of taste :)